openassessment/data.py

```diff
@@ -532,8 +536,16 @@ def collect_ora2_responses(cls, course_id, desired_statuses=None):
             item_id = item['item_id']
             status = item['status']
             result[item_id]['total'] = result[item_id].get('total', 0) + 1
+
             usage_key = BlockUsageLocator.from_string(item_id)
-            result[item_id]['link'] = get_redirect_url(usage_key.course_key, usage_key)
+            link = ""
+            try:
+                link = get_redirect_url(usage_key.course_key, usage_key)
+            except ItemNotFoundError as e:
+                log.exception(str(e))
+            finally:
```

Comment:
Could you remove the `final` block, as it used for clear resources, which was allocated in the try or except blocks

Reply:
Done